### ISSUE TITLE
[Bug] Automatic answer doesn't work when using AnkiDroidJS.ankiTtsSpeak() for TTS (Set TTS language = Don't speak)

### ISSUE THREAD

**Issue by MagTun** [ankidroid/Anki-Android#10766]
In brief: "automatic answer" only works if the "Set TTS language" is not set to "Don't speak" (I have tried with latest alpha version cf. the debug below)

 
In details: I am using a JS script to set the field for the front/back. Sometimes the front field is in English and sometimes it is in Dutch. As the language changes, I can't use the "set TTS language" features and I have set it to "don't speak" for both front and back. For TTS, I use instead `AnkiDroidJS.ankiTtsSpeak()` (cf. below).  With these settings, the automatic answer doesn't work.  The TTS is short (max 4sec) and I have tried with automatic show answer after 9 seconds and automatic next question after 15 seconds. 

However, if I reactivate the "set TTS language" and set it to English and Dutch, then automatic answer works as expected (even with timeout set to 2 and 3 seconds) 

```
AnkiDroidJS.ankiTtsSetLanguage(languageFront);
AnkiDroidJS.ankiTtsSpeak(textforTTS);
```

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".
AnkiDroid Version = 2.16alpha55

Android Version = 10

Manufacturer = Xiaomi

Model = Redmi 7A

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 10; Redmi 7A Build/QKQ1.191014.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/100.0.4896.79 Mobile Safari/537.36

ACRA UUID = 06e6dc69-fef2-4e49-a15a-2b41ca4d7007

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

###### Research
- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by mikehardy** [comment]
@krmanik this one might be interesting for you? It's right in the intersection of the JS API and the Java, with an interesting TTS / automatic answer interaction...

**Comment by krmanik** [comment]
I will look into this.

**Comment by krmanik** [comment]
Even if JS API not added to front/back of card templates, the Automatic Answer doesn't work if `Set TTS language` option set to `Don't speak`.

I have tried to find the cause but didn't find it.
1. First checked NO_TTS, then it calls `onDone` function
https://github.com/ankidroid/Anki-Android/blob/5289e3b8afe43df00e05a1c8c0573c13727854d5/AnkiDroid/src/main/java/com/ichi2/anki/ReadText.kt#L124-L128

2. In `onDone` function, card question/answer displayed 
https://github.com/ankidroid/Anki-Android/blob/5289e3b8afe43df00e05a1c8c0573c13727854d5/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt#L1287-L1296

3. Logcat shows that question is displayed but reviewer still showing answer
```
3708-3708/com.ichi2.anki D/AbstractFlashcardViewer$LinkDetectingGestureDetector: Removing pending touch events for gestures
3708-3708/com.ichi2.anki D/ReadText: ReadText.textToSpeech() method started for string 'Answer 2', locale ''
3708-3708/com.ichi2.anki V/MetaDB: getLanguage: SELECT language FROM languages WHERE did = ? AND ord = ? AND qa = ? LIMIT 1
3708-3708/com.ichi2.anki D/ReadText: ReadText.textToSpeech() method found language choice '0'
3708-3708/com.ichi2.anki D/AbstractFlashcardViewer$ReadTextListener: done reading text
3708-3708/com.ichi2.anki I/AutomaticAnswer: Automatically showing question in 11000ms
3708-3708/com.ichi2.anki I/AutomaticAnswer: stop: automatically show question
3708-3708/com.ichi2.anki D/AbstractFlashcardViewer$CardViewerWebClient: Java onPageFinished triggered: data:text/html;charset=utf-8;base64,
3708-3708/com.ichi2.anki D/AbstractFlashcardViewer$CardViewerWebClient: Java onPageFinished triggered: file:///storage/emulated/0/AnkiDroid/collection.media/__viewer__.html
3708-3708/com.ichi2.anki D/AbstractFlashcardViewer$CardViewerWebClient: New URL, triggering JS onPageFinished: file:///storage/emulated/0/AnkiDroid/collection.media/__viewer__.html
3708-14678/com.ichi2.anki I/com.ichi2.anki: Explicit concurrent copying GC freed 39976(1705KB) AllocSpace objects, 6(260KB) LOS objects, 49% free, 7445KB/14MB, paused 594us total 171.998ms
```

**Comment by krmanik** [comment]
still searching for solutions

**Comment by WiliTest** [comment]
still searching for solutions

**Comment by david-allison** [comment]
Should be resolved in 2.17, let me know if I'm incorrect
